src/main/java/uk/gov/hmcts/reform/sscs/transform/deserialize/SubmitYourAppealToCcdCaseDataDeserializer.java

```diff
@@ -181,22 +184,25 @@ private static String getReasonForBeingLate(SyaCaseWrapper syaCaseWrapper) {
     }
 
     private static String getDwpIssuingOffice(SyaCaseWrapper syaCaseWrapper) {
+        DwpAddressLookupService dwpLookup = new DwpAddressLookupService();
         String benefitType = syaCaseWrapper.getBenefitType().getCode();
-        if (benefitType.equalsIgnoreCase("uc")
-                && (syaCaseWrapper.getMrn() == null
-                || syaCaseWrapper.getMrn().getDwpIssuingOffice() == null)) {
-            return "Universal Credit";
-        } else {
-            if (!mrnIsNotProvided(syaCaseWrapper)) {
-                String dwpIssuingOffice = syaCaseWrapper.getMrn().getDwpIssuingOffice();
-                if (dwpIssuingOffice != null) {
-                    return new DwpAddressLookupService().getDwpMappingByOffice(benefitType, dwpIssuingOffice)
-                            .map(office -> office.getMapping().getCcd())
-                            .orElse(null);
+        String result = null;
+        switch (Benefit.getBenefitByCode(benefitType)) {
+            case UC:
```

Comment:
does sonar allow you to have switch statements now? Sure they used to stop you using them for some reason.

Reply:
yip passes Sonar.